source/agora/network/NetworkManager.d

```diff
@@ -824,6 +826,62 @@ public class NetworkManager
         return format("http://%s:%s", this.node_config.address,
             this.node_config.port);
     }
+
+    /***************************************************************************
+
+        Instantiates a client object implementing `BlockExternalizedHandler`
+
+        In the default implementation, this returns a `BlockExternalizedHandler`
+
+        Params:
+            address = The address (IPv4, IPv6, hostname) of target Server
+        Returns:
+            A BlockExternalizedHandler to communicate with the server
+            at `address`
+
+    ***************************************************************************/
+
+    public BlockExternalizedHandler getBlockExternalizedHandler
+        (Address address)
+    {
+        import vibe.http.client;
+
+        auto settings = new RestInterfaceSettings;
+        settings.baseURL = URL(address);
+        settings.httpClientSettings = new HTTPClientSettings;
+        settings.httpClientSettings.connectTimeout = this.node_config.timeout;
+        settings.httpClientSettings.readTimeout = this.node_config.timeout;
+
+        return new RestInterfaceClient!BlockExternalizedHandler(settings);
+    }
+
+    /***************************************************************************
+
+        Instantiates a client object implementing `PreImageReceivedHandler`
+
+        In the default implementation, this returns a `PreImageReceivedHandler`
+
+        Params:
+            address = The address (IPv4, IPv6, hostname) of target Server
+        Returns:
+            A PreImageReceivedHandler to communicate with the server
+            at `address`
+
+    ***************************************************************************/
+
+    public PreImageReceivedHandler getPreimageReceivedHandler
+        (Address address)
+    {
+        import vibe.http.client;
+
+        auto settings = new RestInterfaceSettings;
+        settings.baseURL = URL(address);
+        settings.httpClientSettings = new HTTPClientSettings;
+        settings.httpClientSettings.connectTimeout = this.node_config.timeout;
+        settings.httpClientSettings.readTimeout = this.node_config.timeout;
+
+        return new RestInterfaceClient!PreImageReceivedHandler(address);
```

Comment:
Don't you need to set `settings` instead of the `address`?

Reply:
Not a single value.
Multiple `PreImageRecoveredHandler settings create a new instance `RestInterfaceClient`.

Reply:
He means you should write this:
```suggestion
        return new RestInterfaceClient!PreImageReceivedHandler(settings);
```

Reply:
Oh, I didn't see it.
That's right. It's wrong.